src/Dapr.AspNetCore/CloudEventsMiddleware.cs

```diff
@@ -97,6 +105,27 @@ private async Task ProcessBodyAsync(HttpContext httpContext, string charSet)
                     contentType = "application/json";
                 }
             }
+            else if (isBinaryDataSet)
+            {
+                // As per the spec, if the implementation determines that the type of data is Binary,
+                // the value MUST be represented as a JSON string expression containing the Base64 encoded
+                // binary value, and use the member name data_base64 to store it inside the JSON object.
+                var decodedBody = Encoding.UTF8.GetString(binaryData.GetBytesFromBase64());
+                try
+                {
+                    JsonDocument.Parse(decodedBody);
+                }
+                catch(JsonException)
+                {
+                    throw new DaprException("Base64 encoded data is not in Json format.");
+                }
+                body = new MemoryStream();
+                var sw = new StreamWriter(body);
+                sw.Write(decodedBody);
+                sw.Flush();
+                body.Seek(0L, SeekOrigin.Begin);
+                contentType = "application/json";
```

Comment:
This doesn't seem right. Remember that the `datacontenttype` field contains a base64 encoded representation of some arbitrary content. 

Consider what happens when you take an image in `.png` format and try to use an event to send it. The cloudevent should look like this:

```json
{
  ...
  "data_base64": "aW1hZ2VfZGF0YQo=",
  "datacontenttype": "application/octet-stream",
  ...
}
```

This should retrieve the bytes (`GetBytesFromBase64`) and write them to the stream.

Also the content type should be set based on `datacontenttype`.

Reply:
You can't make any assumptions of what `aW1hZ2VfZGF0YQo=` means (in this example), that's the whole point of this feature.